Add reference to suggested article

mani2106 opened the pull request:
Added a suggested article to help the reader understand the structure of the decision tree

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
Fixes #13558 
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
Added reference to suggested useful article

#### Any other comments?
None

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

mani2106 committed:
Add reference to article

Added a suggested article to help the reader understand the structure of the decision tree

NicolasHug commented on doc/modules/tree.rst:
```diff
@@ -535,7 +535,9 @@ where :math:`X_m` is the training data in node :math:`m`
     * https://en.wikipedia.org/wiki/Predictive_analytics
-
+    
+    * `Understanding the decision tree structure`<https://scikit-learn.org/dev/auto_examples/tree/plot_unveil_tree_structure.html/>
```
It might be better suited for the `Tips on practical use` section

Also the link should be ``` `Understanding... <http...>`_ ``` (maybe we can reference an example with sphinx-gallery? I'm not sure)

mani2106 replied:
I agree with your suggestion on adding the link to the `Tips on practical use`, thanks for pointing out the misplaced quote. By 

> we can reference an example with sphinx-gallery

Do you mean to say that we can add the output of the referred code or the code blocks

albertcthomas replied:
> It might be better suited for the `Tips on practical use` section

I agree and I would also add a link to the example in the `Example` subsection of the `Classification` section.

NicolasHug replied:
>Do you mean to say that we can add the output of the referred code or the code blocks

I was thinking about using a sphinx ref rather than a hard html link, like ```:ref:`sphx_glr_auto_examples_tree_plot_iris_dtc.py` ```

mani2106 replied:
I have removed it from the Reference and added it to the `Tips on Practical use` section in the latest commit

albertcthomas reviewed (commented):
Thanks for the PR @mani2106

mani2106 committed:
Update tree.rst with sphinx gallery link

Added the suggested article to Examples and Tips on Practical use section, and removed it from references

NicolasHug commented on doc/modules/tree.rst:
```diff
@@ -535,7 +539,7 @@ where :math:`X_m` is the training data in node :math:`m`
 
     * https://en.wikipedia.org/wiki/Predictive_analytics
-
+    
```
Please avoid blank spaces

NicolasHug reviewed (approved):
LGTM!

jnothman commented on doc/modules/tree.rst:
```diff
@@ -345,6 +345,10 @@ Tips on practical use
     give your tree a better chance of finding features that are discriminative.
 
+  * :ref:`sphx_glr_auto_examples_tree_plot_unveil_tree_structure.py` will help
+    in gaining more insights about the nodes which can have an influence over the
```
"how a tree makes predictions" might be clearer than "the nodes which can have an influence ..."

mani2106 committed:
Update tree.rst - modify description

Modified description for the preferred link

jnothman reviewed (approved):
Thanks @mani2106
